Gems/Atom/RPI/Code/Include/Atom/RPI.Public/GpuQuery/GpuPassProfiler.h

```diff
@@ -0,0 +1,132 @@
+            struct PassEntry
+            {
+                //! Total number of attributes collected per  PipelineStatistics.
+                static const uint32_t PipelineStatisticsAttributeCount = 7u;
+
+                using PipelineStatisticsArray = AZStd::array<uint64_t, PipelineStatisticsAttributeCount>;
+
+                PassEntry() = default;
+                PassEntry(const class AZ::RPI::Pass* pass, PassEntry* parent);
+
+                ~PassEntry() = default;
+
+                //! Links the child TimestampEntry to the parent's, and sets the dirty flag for both the parent and child entry.
+                //! Calling this method will effectively add a parent->child reference for this instance, and all parent entries leading up to this
+                //! entry from the root entry.
+                void LinkChild(PassEntry* childEntry);
+
+                //! Checks if timestamp queries are enabled for this PassEntry.
+                bool IsTimestampEnabled() const;
+                //! Checks if PipelineStatistics queries are enabled for this PassEntry.
+                bool IsPipelineStatisticsEnabled() const;
+
+                //! The name of the pass.
+                Name m_name;
+                //! Cache the path name of the Pass as a unique identifier.
+                Name m_path;
+
+                RPI::TimestampResult m_timestampResult;
+                uint64_t m_interpolatedTimestampInNanoseconds = 0u;
+
+                //! Convert the PipelineStatistics result to an array for easier access.
+                PipelineStatisticsArray m_pipelineStatistics;
+
+                //! Used as a double linked structure to reference the parent <-> child relationship.
+                PassEntry* m_parent = nullptr;
+                AZStd::vector<PassEntry*> m_children;
+
+                //! Mirrors the enabled queries state of the pass.
+                bool m_timestampEnabled = false;
+                bool m_pipelineStatisticsEnabled = false;
+
+                //! Mirrors the enabled/disabled state of the pass.
+                bool m_enabled = false;
+
+                //! Dirty flag to determine if this entry is linked to an parent entry.
+                bool m_linked = false;
+
+                //! Cache if the pass is a parent.
+                bool m_isParent = false;
+            };
+
+            GpuPassProfiler() = default;
+            ~GpuPassProfiler() = default;
+
+            GpuPassProfiler(GpuPassProfiler&& other) = delete;
+            GpuPassProfiler(const GpuPassProfiler& other) = delete;
+            GpuPassProfiler& operator=(const GpuPassProfiler& other) = delete;
+
+            void SetRenderPipelineGpuTimeMeasurementEnabled(bool enabled) { m_measureRenderPipelineGpuTime = enabled; }
+            bool IsRenderPipelineGpuTimeMeasurementEnabled() { return m_measureRenderPipelineGpuTime; }
```

Comment:
Suggest: remove RenderPipeline in the function names.

Reply:
Done.